src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -851,9 +851,9 @@ module Crystal
              (@block_nest > 0)
             ivar = scope.lookup_instance_var(var_name)
             if found_self
-              ivar.nil_reason = NilReason.new(var_name, :used_self_before_initialized, found_self)
+              ivar.nil_reason = NilReason.new(var_name, NilReason::Reason::UsedSelfBeforeInitialized, found_self)
```

Comment:
If you put a type restriction Reason on the reason argument of NilReason, then you can pass a symbol here which the compiler will change to that enum member. Then it stops being overly verbose :-)

Reply:
Huh, I'd heard symbols could be auto-casted to enum members but couldn't get it to work, thank you for teaching me!